libraries/chain/sps_evaluator.cpp

```diff
@@ -74,6 +74,9 @@ void update_proposal_votes_evaluator::do_apply( const update_proposal_votes_oper
    {
       FC_ASSERT( _db.has_hardfork( STEEM_PROPOSALS_HARDFORK ), "Proposals functionality not enabled until hardfork ${hf}", ("hf", STEEM_PROPOSALS_HARDFORK) );
 
+      const auto& account = _db.get_account( o.voter );
+      FC_ASSERT( account.can_vote, "Account has declined its voting rights." );
```

Comment:
This should be wrapped in a hardfork check, otherwise an account with declined voting rights who attempts to vote for an SPS proposal before this hardfork takes effect would cause issues.

Reply:
Good point @Smittyvb. I fixed it.
Thanks for reviewing the code.